core/job/job.go

```diff
@@ -0,0 +1,22 @@
+package job
+
+import (
+	"context"
+	"time"
+)
+
+type Repository interface {
+	GetSyncJobsByService(ctx context.Context, serviceName string) ([]JobsQueue, error)
+}
+
+type JobsQueue struct {
+	ID          string    `db:"id"`
+	Type        string    `db:"type"`
+	LastError   string    `db:"last_error"`
+	AttemptsDo  int32     `db:"attempts_do"`
+	Payload     []byte    `db:"payload"`
+	RunAt       time.Time `db:"run_at"`
+	CreatedAt   time.Time `db:"created_at"`
+	UpdatedAt   time.Time `db:"updated_at"`
+	LastAttempt time.Time `db:"last_attempt"`
+}
```

Comment:
Can you add some documentation in the issue if possible how this job flow comes into picture. I'm assuming this is to ensure there's only one instance of the job per index...

Reply:
yes correct, we will only allow 1 job for 1 index at the same time. Will update the issue